docs/reference/garbage-collection.md

```diff
@@ -394,3 +393,71 @@ spark-submit --class io.treeverse.clients.GCBackupAndRestore \
   objects-to-restore-list-location backup-external-location storage-namespace azure
   ```
 </div>
+
+## Beta: Deleting uncommitted objects
+
+Run this job to [delete objects that were never committed and have no valid pointers from lakeFS](https://github.com/treeverse/lakeFS/issues/1933). This may happen when:
+1. Objects are deleted or overwritten.
+2. Changes are reverted from a branch before being committed.
+3. A branch with uncommitted changes is deleted.
+
+The uncommitted GC will not clean:
+1. Objects that are included in any commit. For committed objects cleanup see [above](#what-gets-collected)
+2. Everything mentioned in [what does not get collected](#what-does-_not_-get-collected)
+
+To run the uncommitted GC job run:
+  ```bash
+spark-submit \
+    --conf spark.hadoop.lakefs.api.url<LAKEFS_ENDPOINT> \
+    --conf spark.hadoop.fs.s3a.access.key=<AWS_ACCESS_KEY_ID> \
+    --conf spark.hadoop.fs.s3a.secret.key=<AWS_SECRET_ACCESS_KEY> \
+    --conf spark.hadoop.lakefs.api.access_key=<LAKEFS_ACCESS_KEY_ID> \
+    --conf spark.hadoop.lakefs.api.secret_key=<LAKEFS__SECRET_ACCESS_KEY> \
+    --class io.treeverse.gc.UncommittedGarbageCollector \
+    --packages org.apache.hadoop:hadoop-aws:2.7.7,\
+          lakefs-spark-client-<VERSION>-assembly-<TODO>.jar \
+    <REPO_NAME> <TODO_OUTPUT_LOCATION>
+```
+
+#### Uncommitted GC job options
+
+Similar to the described [above](#gc-job-options).
+
+#### Backup and restore 
+
+<TODO> - will it work?
```

Comment:
It will work, but lets put a pin on that for the first milestone

Reply:
Just like mark&sweep - it's a safety mechanisms that users can rely on for safer delete. It can wait though.

Reply:
So let's change the comment to 'be supported in the future'